### PR TITLE
nautilus: rpm: add python3-saml as install dependency

### PR DESCRIPTION
**Pull request by epuertat**
backport tracker: https://tracker.ceph.com/issues/44997

---

backport of https://github.com/ceph/ceph/pull/26819 https://github.com/ceph/ceph/pull/33229 https://github.com/ceph/ceph/pull/34381 
parent tracker: https://tracker.ceph.com/issues/44721

this backport was staged using ceph-backport.sh version 15.1.1.389
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by epuertat** [comment]
@smithfarm please check if you are ok with having no guards around this dep, same way as in master or octopus. Thanks!

File: ceph.spec.in [hunk 1]
```diff
@@ -484,12 +484,16 @@ Requires:       python%{_python_buildid}-cherrypy
 Requires:       python%{_python_buildid}-jwt
 Requires:       python%{_python_buildid}-routes
 Requires:       python%{_python_buildid}-werkzeug
+%if 0%{?weak_deps}
```
Comment (tchaikov) [inline ceph.spec.in:487]: probably we need to backport 147531515b51d7b571b3a4a97e7cbdce239a049a as well, for this macro.

File: ceph.spec.in [hunk 2]
```diff
@@ -484,12 +484,16 @@ Requires:       python%{_python_buildid}-cherrypy
...
+Recommends:     python%{_python_buildid}-saml
+%endif
 %endif
 %if 0%{?suse_version}
 Requires:       python%{_python_buildid}-CherryPy
 Requires:       python%{_python_buildid}-PyJWT
 Requires:       python%{_python_buildid}-Routes
 Requires:       python%{_python_buildid}-Werkzeug
+Recommends:     python%{_python_buildid}-python3-saml
```
Comment (smithfarm) [inline ceph.spec.in:496]: Please drop this line entirely - nautilus runs on SLE-15-SP1 which does not have this package.

File: ceph.spec.in (hunk 1 above)
Comment (epuertat) [reply ceph.spec.in:491]: Done! Thanks!

File: ceph.spec.in (hunk 2 above)
Comment (epuertat) [reply ceph.spec.in]: Done!

**Comment by epuertat** [comment]
Extra backport needed (for RHEL8): https://github.com/ceph/ceph/pull/33229

**Comment by epuertat** [comment]
[Packages building...](https://shaman.ceph.com/builds/ceph/wip-44997-nautilus/)

**Comment by yuriw** [comment]
[wip-yuri6-testing-2020-05-13-1919-nautilus](https://trello.com/c/JTjhE7Sj/903-wip-yuri6-testing-2020-05-13-1919-nautilus)

**Pull request merged**